### PR TITLE
Eco 1406/remove top bar web exp

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Support new UI for webview
#### Technical description:
- Removed Toolbar from poll webview
- Send new param: `KinTheme` to the webview

### CODE REVIEW

File: sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import com.kin.ecosystem.core.bi.events.EarnOrderCompleted.OfferType
+import com.kin.ecosystem.core.bi.events.EarnOrderFailed.Origin
+import com.kin.ecosystem.core.data.internal.Configuration
+import com.kin.ecosystem.core.data.order.OrderDataSource
+import com.kin.ecosystem.core.network.model.OpenOrder
+import com.kin.ecosystem.core.network.model.Order
+import com.kin.ecosystem.poll.view.IPollWebView
+import com.kin.ecosystem.poll.view.IPollWebView.Message
+
+class PollWebViewPresenter(private val pollJsonString: String, private val offerID: String,
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt:18]: do we have any functionality change in this file or it just transformation to kotlin?
if yes can you pinpoint to the changes?

File: sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    private fun initWebView() {
+        view?.initWebView()
+    }
+
+    private fun createOrder() {
+        try {
+            eventLogger.send(EarnOrderCreationRequested
+                    .create(EarnOrderCreationRequested.OfferType.fromValue(contentType), amount.toDouble(), offerID, EarnOrderCreationRequested.Origin.MARKETPLACE))
+        } catch (ex: IllegalArgumentException) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt]: catch clause is not required in kotlin and can be removed (even for checked exception i.e. non runtime exception)
Actually wondering why in the first place there's a catch as `IllegalArgumentException` is runtime exception.

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebView.java [hunk 3]
```diff
@@ -83,6 +84,27 @@ public void run() {
...
+                    setTheme(kinTheme);
+                }
+            });
+
+            return;
+        }
+
+        final StringBuilder js = new StringBuilder("kin.setTheme(\"").append(kinTheme).append("\")");
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebView.java]: can be extracted to a shared method (with render logic)

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebView.java [hunk 4]
```diff
@@ -83,6 +84,27 @@ public void run() {
...
+            evaluateJavascript(js.toString(), null);
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebView.java]: a potential security risk, make sure input is safe (either by making sure the code path is safe or even more paranoidly by sanitizing the input from js input)

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemNativeApi.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import com.kin.ecosystem.core.Logger
+
+internal class EcosystemNativeApi {
+
+    private var listener: EcosystemWebPageListener? = null
+
+    @JavascriptInterface
+    fun loaded() {
+        Logger.log(Log().withTag(TAG).text("loaded()"))
+        if (listener != null) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/web/EcosystemNativeApi.kt]: prettify the code using kotlin magic to ensure null safety here

**Review by yosriz — COMMENTED** [review]
small comments, waiting for details what logic had changed when migrating from  java to kotlin

File: sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt (hunk 1 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt:18]: regarding hide or show the toolbar + change the title.
all this was removed

File: sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt (hunk 2 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/poll/presenter/PollWebViewPresenter.kt]: not sure but, yohay added it for safety regarding `EarnOrderCreationRequested.OfferType.fromValue(contentType)`

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebView.java (hunk 4 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebView.java]: will do on another task

**Review by yosriz — APPROVED** [review]
👍

**Pull request merged**